### PR TITLE
Refactor Request Password react component

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

It refactors Request Password react component to make it match our standards.
Needed for https://github.com/3scale/porta/pull/1765

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-4619

Tasks:

- [x] Refactor component
- [x] Fix js tests

### CODE REVIEW

File: app/javascript/packs/RequestPasswordPage.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import '@babel/polyfill'
+import 'core-js/es7/object'
+import { isBrowserIE11 } from 'utilities/ie11Utils'
+
+const isIE11 = isBrowserIE11(window)
+if (isIE11) {
```
Comment (didierofrivia) [inline app/javascript/packs/RequestPasswordPage.js]: Maybe the entire conditional could be part of `utilities/ie11Utils` ?

Comment (didierofrivia) [reply app/javascript/packs/RequestPasswordPage.js]: you could export directly a boolean `isIE11`

Comment (didierofrivia) [reply app/javascript/packs/RequestPasswordPage.js]: Different PR better, out of scope... but I've seen it in at least 3 different componenets

Comment (damianpm) [reply app/javascript/packs/RequestPasswordPage.js]: Yes, we can do it in it's own PR

**Comment by didierofrivia** [comment]
https://github.com/3scale/porta/pull/1765 broke that cuke failing

File: app/javascript/packs/RequestPasswordPage.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import '@babel/polyfill'
+import 'core-js/es7/object'
+import { isBrowserIE11 } from 'utilities/ie11Utils'
+
+const isIE11 = isBrowserIE11(window)
```
Comment (josemigallas) [inline app/javascript/packs/RequestPasswordPage.js]: We don't support ie11 anymore

File: app/javascript/packs/RequestPasswordPage.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+if (isIE11) {
+  // eslint-disable-next-line no-unused-expressions
+  import('LoginPage/assets/styles/ie11-pf4BaseStyles.css')
+}
+
+import { RequestPasswordWrapper } from 'LoginPage'
```
Comment (josemigallas) [inline app/javascript/packs/RequestPasswordPage.js]:
```suggestion
import { RequestPasswordWrapper as RequestPassword } from 'LoginPage'
```

File: app/javascript/src/LoginPage/RequestPasswordWrapper.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// @flow
```
Comment (josemigallas) [inline app/javascript/src/LoginPage/RequestPasswordWrapper.jsx]: The name of the file should be simply `RequestPassword.jsx`.

File: app/javascript/src/LoginPage/RequestPasswordWrapper.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+const RequestPassword = ({ flashMessages, providerLoginPath, providerPasswordPath }: Props) => (
+  <LoginPage
+    brandImgSrc={brandImg}
+    brandImgAlt="Red Hat 3scale API Management"
+    backgroundImgSrc={PF4DownstreamBG}
+    backgroundImgAlt="Red Hat 3scale API Management"
+    loginTitle={`Request Password`}
+    footer={null}
```
Comment (josemigallas) [inline app/javascript/src/LoginPage/RequestPasswordWrapper.jsx]: Is it possible not to pass this prop? (is it default to null?)

File: app/javascript/src/LoginPage/RequestPasswordWrapper.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  providerPasswordPath: string
+}
+
+const RequestPassword = ({ flashMessages, providerLoginPath, providerPasswordPath }: Props) => (
+  <LoginPage
+    brandImgSrc={brandImg}
+    brandImgAlt="Red Hat 3scale API Management"
+    backgroundImgSrc={PF4DownstreamBG}
+    backgroundImgAlt="Red Hat 3scale API Management"
+    loginTitle={`Request Password`}
```
Comment (josemigallas) [inline app/javascript/src/LoginPage/RequestPasswordWrapper.jsx]:
```suggestion
    loginTitle="Request Password"
```

File: app/views/provider/request_password_resets/new.html.slim [hunk 7]
```diff
@@ -1,8 +1,7 @@
-= javascript_pack_tag 'ResetPasswordPage'
+= javascript_pack_tag 'RequestPasswordPage'
 - authentication_providers = (@authentication_providers || []).map { |ap| {authorizeURL: ap.authorize_url, humanKind: ap.human_kind} }
 - flash_messages = (flash || []).map { |f| {type: f[0], message: f[1]}}
-- is_master_account = domain_account.master?
 
-div#pf-reset-page-container data-reset-props={flashMessages: flash_messages,
+div#pf-request-page-container data-request-props={flashMessages: flash_messages,
```
Comment (josemigallas) [inline app/views/provider/request_password_resets/new.html.slim]: IMO it's better with the args aligned, like this:
```suggestion
div#pf-request-page-container data-request-props={
  flashMessages: flash_messages,
```

File: app/javascript/packs/RequestPasswordPage.js (hunk 2 above)
Comment (damianpm) [reply app/javascript/packs/RequestPasswordPage.js]: I know, but I'm not sure if we should remove all IE11 related stuff from everywhere, so I just leave it like the other forms

Comment (damianpm) [reply app/javascript/packs/RequestPasswordPage.js]: Removing!

File: app/javascript/src/LoginPage/RequestPasswordWrapper.jsx (hunk 4 above)
Comment (damianpm) [reply app/javascript/src/LoginPage/RequestPasswordWrapper.jsx]: The other forms are called `LoginPageWrapper` or `SignupPageWrapper` and I didn't want to use an inconsistent name

File: app/javascript/src/LoginPage/RequestPasswordWrapper.jsx (hunk 5 above)
Comment (damianpm) [reply app/javascript/src/LoginPage/RequestPasswordWrapper.jsx]: You are right, default is null

File: app/views/provider/request_password_resets/new.html.slim (hunk 7 above)
Comment (damianpm) [reply app/views/provider/request_password_resets/new.html.slim]: But that's a syntax error in slim

**Pull request merged**